core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1387,6 +1387,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * Register an RDD to be persisted in memory and/or disk storage
    */
   private[spark] def persistRDD(rdd: RDD[_]) {
+    _executorAllocationManager.foreach { _ =>
+      logWarning(
+        s"Dynamic allocation currently does not support cached RDDs. Cached data for RDD " +
+        s"${rdd.id} will be deallocated when executors are removed.")
```

Comment:
I thought about lost but thought it would be too strong, since the data can be recomputed. But sure, it sounds more correct than "deallocated".